frontends/verilog/verilog_parser.y

```diff
@@ -221,6 +224,69 @@ static void rewriteAsMemoryNode(AstNode *node, AstNode *rangeNode)
 	node->children.push_back(rangeNode);
 }
 
+static void fixId(AstNode* node, std::string& loop_id, std::string& id_localscope_name) {
+	if (node) {
+		if (node->str == loop_id) node->str = id_localscope_name;
+		for (auto& itr : node->children) fixId(itr, loop_id, id_localscope_name);
+	}
+}
+
+static void enterScope(std::string name) {
+	if(name.find("unnamed::block") != std::string::npos) {
+		name += std::to_string(block_counter++);
+	}
+	scope_variables.push_back(std::make_pair(name, std::set<std::string>()));
+}
+
+static void addVariable(std::string vName) {
+	if (scope_variables.size() > 0) {
+		scope_variables.back().second.insert(vName);
+	}
+}
+
+
+static void addLocalScope(AstNode *node) {
+	if(!node)
+		return;
+	if(scope_variables.size() == 0 || scope_variables.back().first.find("unnamed::block") == std::string::npos)
+		return;
+	auto scope = scope_variables.back();
+	auto scope_name = scope.first;
+	auto scope_variables = scope.second;
+	std::vector<AstNode*> childrens_to_remove;
+	for (auto *child : node->children) {
+		addLocalScope(child);
+		if(!scope_variables.count(child->str)) //check if variable should be converted to local scope
+			continue;
+		std::string wire_str = child->str;
+		std::string rename_str = scope_name + "::" + wire_str.substr(1);
+
+		fixId(node, wire_str, rename_str);
+		if(child->type == AST_IDENTIFIER) // do not create AST_WIRE in global scope for AST_IDENTFIERs
+			continue;
+
+		log_assert(child->type == AST_WIRE || child->type == AST_GENVAR);
+		AstNode *wire = child->clone();
+		if(node->type == AST_BLOCK)
+			childrens_to_remove.push_back(child);
+		else
+			child->type = AST_IDENTIFIER;
+
+		for (auto itr = ast_stack.rbegin(); itr != ast_stack.rend(); itr++) {
+			auto *node = *itr;
+			if (node->type == AST_MODULE || node->type == AST_FUNCTION) {
+				node->children.push_back(wire);
+				break;
+			}
+		}
+	}
+	for(auto child : childrens_to_remove)
+		node->children.erase(std::remove(node->children.begin(), node->children.end(), child), node->children.end());
+}
```

Comment:
This PR adds quite a bit of complexity to the parser. My initial thought would be to store the declaration and/or initialization as the initial child(ren) of the node, and to have the combined `simplify()` logic for `AST_FOR` and `AST_GENFOR` handle the elaboration and scoping explicitly. Are there hidden issues with that strategy?

Reply:
This is true, I removed the logic from parser and moved it to ``simplify``. It should be much simpler now. I also removed logic for ``unnamed_block`` as support for this is already in master.